h/models/group.py

```diff
@@ -152,6 +169,50 @@ def _write_principal(group):
     }.get(group.writeable_by)
 
 
+class _GroupTypeFlags(object):
+    """Abstract base class for group type flags classes."""
+
+    def __eq__(self, other):
```

Comment:
This works, although if you just want a type that stores some named fields and behaves like a value class, `collections.namedtuple` is the easiest way to get that.

Here you are ignoring the type when comparing for equality - I'm not clear on whether this is by design (it wouldn't be completely unreasonable) or accidental.

Reply:
The thing about `Open|PrivateGroupTypeFlags` is that (because of the `__eq__()`) you can compare them to a `models.Group` and it'll return `True` if it's a group of that type: `is_private = some_group == PrivateGroupTypeFlags()`. The `Group.type` property makes use of this convenience. I don't think a namedtuple would give you that convenience behaviour.

This `__eq__()` is what's actually used to implement the `Group.type` property, so you couldn't use `Group.type` in `__eq__()`. `Open|PrivateGroupTypeFlags.type_` is just used by the `Group.type` property in order to know what string to return.

Ugh, the whole thing is horrible! Trying to pretend we have a single group class when we have two. As I said I think we should put up with that for now. Maybe there is a better way than these TypeFlags classes though, not sure.

Possibly just using namedtuples would be less confusing.